src/prices/updater.ts

```diff
@@ -107,15 +118,14 @@ export function createPriceUpdater(
         return;
       }
 
-      const getBlockTimestamp = async (blockNumber: number) => {
-        const block = await withCacheMaybe(
-          `block-${config.chain.id}-${blockNumber}`,
-          () => provider.getBlock(blockNumber)
-        );
+      const getBlockTimestamp = async (blockNumber: bigint) => {
+        const block = await provider.getBlock(Number(blockNumber));
```

Comment:
Querying the cache here and falling back to the provider if block is not cached would avoid pushing the caching concern down to `getBlockFromTimestamp`.